### DISCUSSION TITLE
Is it possible to add descriptions to path and/or query parameters?

### DISCUSSION THREAD

**Discussion by GeoSander** [vitalik/django-ninja#322] [Q&A] [answered]
Hi,

First of all: thanks for creating and maintaining this project - I like it a lot more than the "regular" Django REST framework!

I was wondering if there is a way to add descriptions to path and query parameters, so that they show up in the Swagger UI?

I tried the FastAPI approach for this, since the same parameter functions seem to be available in Ninja too, for example:

```python
@api.post("tasks/{serial}/{task_id}/map", summary="bla bla", tags=["tasks", "maps"])
def upload_map(request,
               serial: str = Path(..., description="The device serial number."),
               task_id: uuid.UUID = Path(..., description="The UUID of the task to which the map belongs."),
               overwrite: bool = Query(False, description="Overwrite an existing map.")):
    """ Uploads the map data for the given device and task."""
    return {
        "device": serial,
        "task": task_id,
        "overwrite": overwrite
    }
```

Django runs fine like this, but there are still no descriptions in the Swagger UI:

![image](https://user-images.githubusercontent.com/8972335/149358847-77cef8f3-2fe8-44a0-a069-252d52698fb6.png)

Is this even possible? Or do I need to edit the default Swagger HTML template for the descriptions to show up?

Thanks,
Sander

**Comment by vitalik** [comment, marked as answer]
Hi @GeoSander 

this looks like a bug (#326)

**Reply by GeoSander** [reply]
Thanks for sorting that out @vitalik. I was too lazy 😉
Will you fix it or should I try and make a PR?

**Comment by vitalik** [comment]
@GeoSander fixed in 0.16.2

**Reply by GeoSander** [reply]
Great, thanks! 🙏

**Comment by jonesnc** [comment]
I have a question about this snippet:

`Path(..., description="The device serial number.")`

Could someone explain what `...` does?

**Comment by vitalik** [comment]
@jonesnc 
`...` is a special python constant [Ellipsis](https://docs.python.org/dev/library/constants.html#Ellipsis)

in case of Django ninja it is used to mark that this field is required (does not have default value)

**Reply by jonesnc** [reply]
Thanks!
